src/Workspaces/CSharp/Portable/Formatting/Rules/SuppressFormattingRule.cs

```diff
@@ -286,6 +289,109 @@ private void AddStatementExceptBlockSuppressOperations(List<SuppressOperation> l
             AddSuppressWrappingIfOnSingleLineOperation(list, firstToken, lastToken);
         }
 
+        private void AddFormatSuppressOperations(List<SuppressOperation> list, SyntaxNode node)
+        {
+            foreach (var child in node.ChildNodesAndTokens())
+            {
+                if (!child.IsToken)
+                {
+                    continue;
+                }
+
+                ProcessTriviaList(list, child.AsToken().LeadingTrivia);
+                ProcessTriviaList(list, child.AsToken().TrailingTrivia);
+            }
+
+            // Local functions
+            static void ProcessTriviaList(List<SuppressOperation> list, SyntaxTriviaList triviaList)
+            {
+                foreach (var trivia in triviaList)
+                {
+                    ProcessTrivia(list, trivia);
+                }
+            }
+
+            static void ProcessTrivia(List<SuppressOperation> list, SyntaxTrivia trivia)
+            {
+                if (!(trivia.HasStructure))
+                {
+                    return;
+                }
+
+                ProcessStructuredTrivia(list, trivia.GetStructure());
+            }
+
+            static void ProcessStructuredTrivia(List<SuppressOperation> list, SyntaxNode structure)
+            {
+                if (!(structure is PragmaWarningDirectiveTriviaSyntax pragmaWarningDirectiveTrivia))
+                {
+                    return;
+                }
+
+                if (!IsFormatDirective(pragmaWarningDirectiveTrivia, SyntaxKind.DisableKeyword))
+                {
+                    return;
+                }
+
+                var startToken = pragmaWarningDirectiveTrivia.GetLastToken(includeZeroWidth: true, includeDirectives: true);
+
+                var endDirective = pragmaWarningDirectiveTrivia.GetNextDirective(trivia => IsFormatDirective(trivia, SyntaxKind.RestoreKeyword));
+                var endToken = endDirective is null
+                    ? structure.SyntaxTree.GetRoot().GetLastToken(includeZeroWidth: true)
```

Comment:
Can cast to CompilationUnitSyntax and then just grab the EOF token.

Reply:
Are there any cases where this would not work? e.g. formatting a partial tree or formatting a script?

Reply:
I don't believe so.  Indeed, it's *the* most reliable way to get to the end/full-span of the entire tree.

Reply:
note: i don't feel strongly about this. if your above code works well for all sorts of paranoid cases, i'm fine with it :)

Reply:
Updated!